### PR TITLE
Support loading non-string models from memory

### PR DESCRIPTION
**Pull request by pnehrer**
We've run into an issue with `Core::read_model_from_buffer` -- the wrapper assumes that the in-memory model is a valid string. However, non-native models, such as ONNX, aren't valid strings. Attempting to load an ONNX model results in an error creating a CString due to non-terminating null characters.

We propose to change the method interface to support arbitrary byte models rather than just strings.

Furthermore, we propose to make the weight tensor an optional argument, as it doesn't apply to non-native models.

### CODE REVIEW

File: crates/openvino/src/core.rs [hunk 1]
```diff
@@ -51,17 +53,19 @@ impl Core {
     }
 
     /// Read model with model and weights loaded in memory.
-    pub fn read_model_from_buffer(
+    pub fn read_model_from_buffer<'a, T: Into<Option<&'a Tensor>>>(
         &mut self,
-        model_str: &str,
-        weights_buffer: &Tensor,
+        model_str: &[u8],
+        weights_buffer: T,
```
Comment (abrown) [inline crates/openvino/src/core.rs:59]: I propose we move back to the `Option<&Tensor>` like you had; it's a bit more straightforward and avoids potential concerns with code size, etc.

**Review by abrown — APPROVED** [review]
Thanks for the contribution; I think it makes sense. Can you fix the one comment below before I merge?

**Comment by abrown** [comment]
Oh, and one other thing: the CI failures are not due to this change. I think you will have to rebase to see CI turn green.

File: crates/openvino/src/core.rs (hunk 1 above)
Comment (pnehrer) [reply crates/openvino/src/core.rs:59]: Sounds good 👍

**Comment by pnehrer** [comment]
> Oh, and one other thing: the CI failures are not due to this change. I think you will have to rebase to see CI turn green.

I think this is a function that isn't in 2022.3.0, which is what seems to be failing to link. Some more recent version jobs do succeed before this one cancels the rest.

**Comment by pnehrer** [comment]
> > Oh, and one other thing: the CI failures are not due to this change. I think you will have to rebase to see CI turn green.
> 
> I think this is a function that isn't in 2022.3.0, which is what seems to be failing to link. Some more recent version jobs do succeed before this one cancels the rest.

... and as a result, I can remove the unit test if that helps.

**Comment by abrown** [comment]
Ah, I see: @rahulchaphalkar and I noticed a failure in the Windows builds due to the missing function and I chalked it up to some  difference between the Windows and Linux releases. But actually we were never exercising the function so it would have failed if someone actually tried it. Given that, I propose we keep the test and remove the following step in CI since we just can't support those versions with this function:

https://github.com/intel/openvino-rs/blob/3a404eef023e854b592528e98e6b3b759735bff3/.github/workflows/main.yml#L28-L32

With that out of the way, I think this _should_ be good to go 🤞

**Comment by pnehrer** [comment]
Would you still want to keep the "oldest supported version" with apt and just pull it up to 2023.2.0, instead of removing that combo altogether?

**Comment by abrown** [comment]
> Would you still want to keep the "oldest supported version" with apt and just pull it up to 2023.2.0, instead of removing that combo altogether?

Well, we have 2023.2.0 up above in the non-APT build so I'm not too worried about it.

**Pull request merged**